data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/SqsQueueUrl.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class SqsQueueUrl {
+    private final URL queueUrl;
+    private final String accountId;
+
+    private SqsQueueUrl(final URL queueUrl) throws MalformedURLException {
+        this.queueUrl = queueUrl;
+        final String path = queueUrl.getPath();
+
+        if (path.isEmpty())
+            throw new MalformedURLException();
+
+        final String[] pathParts = path.split("/");
+        if (pathParts.length < 3)
+            throw new MalformedURLException();
+
+        accountId = pathParts[1];
```

Comment:
Let's confirm this is a 12 digit number.

Reply:
I wrote this class more for the purpose of getting the `accountId` as supplied by the user rather than trying to provide a full validation on the SQS queue. But, I can make this validate the accountId is correct per the spec.